pkg/providers/generic.go

```diff
@@ -0,0 +1,103 @@
+package providers
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/marcosnils/bin/pkg/assets"
+)
+
+type generic struct {
+	baseURL   string
+	name      string
+	version   string
+	os        string
+	arch      string
+	ext       string
+	latestURL string
+}
+
+func (g *generic) GetID() string {
+	return "generic"
+}
+
+func (g *generic) GetLatestVersion() (string, string, error) {
+	log.Debugf("Getting latest release for %s", g.name)
+
+	resp, err := http.Get(g.latestURL)
```

Comment:
we should check if `latestURL` is not empty here so we can add proper `debug` logs for this with a message like: `latest url not found for $name` or something